packages/artifact/src/internal/path-and-artifact-name-validation.ts

```diff
@@ -0,0 +1,82 @@
+import {info} from '@actions/core'
+
+/**
+ * Invalid characters that cannot be in the artifact name or an uploaded file. Will be rejected
+ * from the server if attempted to be sent over. These characters are not allowed due to limitations with certain
+ * file systems such as NTFS. To maintain platform-agnostic behavior, all characters that are not supported by an
+ * individual filesystem/platform will not be supported on all fileSystems/platforms
+ *
+ * FilePaths can include characters such as \ and / which are not permitted in the artifact name alone
+ */
+const invalidArtifactFilePathCharacters = new Map<string, string>([
+  ['"', 'Double quote "'],
+  [':', ' Colon :'],
+  ['<', ' Less than <'],
+  ['>', ' Greater than >'],
+  ['|', ' Vertical bar |'],
+  ['*', ' Asterisk *'],
+  ['?', ' Question mark ?'],
+  ['\r', ' Carriage return \\r'],
+  ['\n', ' Line feed \\n']
+])
+
+const invalidArtifactNameCharacters = new Map<string, string>([
+  ...invalidArtifactFilePathCharacters,
+  ['\\', ' Backslash \\'],
```

Comment:
Is the backslash escaped for a specific reason? And do we need to escape it for the display message?

Reply:
Yes, this has to be escaped. In general it's a special character so if you want to display it or literally express it then you need `\\`

![image](https://user-images.githubusercontent.com/16109154/144302470-5a1dd75b-7a1d-436d-bc27-d60e44e34eb5.png)

Reply:
oops, forgot 🤦